### ISSUE TITLE
[Feedback and a few minor issues] Great work, few issues with restricted movement

### ISSUE THREAD

**Issue by blackwatersgit** [KozGit/DOOM-3-BFG-VR#195]
Hi KozGit, 

First of I would like to address that this mod is amazing. It implements VR and touch better than most actual VR titles. With the features like  the options of free locomotion and I must admit I don't even feel motionsick after playing for a few hours.

There's a couple of things I have experienced using the mod that you might be interested in.

1. Using the PDA when watching videos you can still move to the other tabs (email etc.) where audio will be playing. But moving back to the video after being in any other tab will reset the video. I would prefer it continued on playing in this case.

2. Leaning around corners feels weird. I think this has to do with the implementation of moving into restricted areas like through walls using "roomscale" walking in the real world. Sometimes leaning you will be moved so you actually have to move with the controller again to regain the advantageous position on the enemies. 

Is there any option to disable this restriction so it'd even be possible to walk into walls? Not that I would use it, but it's a quick fix for the leaning.

3. Using terminals and screens sometimes I will be moved backwards while reaching my arm to the button on screen. I think this has to do with the restricted movement, or maybe that I have a small play area when standing. 

4. Using a body model instead of just hands often lets me see through the body. I think you already said you can fix this by making the body orientate according to the headsets direction instead of the last moved direction. I, too, prefer having a body with arms compared to the floating hands.

If you would like a video of any of this I will happily try to record some gameplay where I experience this.

Like I said these are minor issues but I can only tell you I appreciate you work on this project. No VR game yet have been entertaining me like this mod, it feels so naturally implemented that I am amazed by it every time I launch up the game.

Oh and by the way, did you give up on the RTCW source code to add in VR?

**Comment by KozGit** [comment]
Glad you like the mod!

The Video/PDA replay is just the way the PDA menus were originally designed, maybe later I can take a look at the video restart, but to be honest it will be a low priority until I can finish/fix other issues.

In regards to leaning, it will always be an issue due to the fact that the game has no way to know where your waist is in real life.  If there was a third tracking point on the waist, lots of things would be better.  I suppose it would be possible to add a 'lean' button, something I'll have to think about. Walking through walls would cause all kinds of problems with setting off game triggers, so I don't forsee it being enabled by default.

It would appear that there is an issue with the current 'lean' code in regards to reaching out to press buttons on in game guis - it's something I'm actually working on right now.

I haven't had the time to finish this yet, so I really haven't had time to do anything other than briefly glance at the RTCW code.  Once this is finished tho, I do want to start on either RTCW or Tron 2.0 if I can get it going. ( So I wouldn't say I've given up - just haven;t even had a chance to start yet :) )
